openstack/images_image_v2.go

```diff
@@ -127,7 +129,30 @@ func resourceImagesImageV2File(client *gophercloud.ServiceClient, d *schema.Reso
 
 			defer resp.Body.Close()
 
-			if _, err = io.Copy(file, resp.Body); err != nil {
+			var reader io.ReadCloser
+			decompress := d.Get("decompress").(bool)
+
+			if decompress && !resp.Uncompressed {
+				// If we're here "Content-Encoding" should be not existant,
+				// we'll read "Content-Type" to select format
+				switch resp.Header.Get("Content-Type") {
+				case "gzip", "application/gzip":
+					reader, err = gzip.NewReader(resp.Body)
+					if err != nil {
+						return "", fmt.Errorf("Error decompressing gzip image: %s", err)
+					}
+				case "bzip2", "application/bzip2", "application/x-bzip2":
+					bz2_reader := bzip2.NewReader(resp.Body)
+					reader = io.NopCloser(bz2_reader)
```

Comment:
Let's follow the golang var name standards.
```suggestion
					bz2Reader := bzip2.NewReader(resp.Body)
					reader = io.NopCloser(bz2Reader)
```